cpp/include/cudf/aggregation.hpp

```diff
@@ -259,16 +261,75 @@ std::unique_ptr<Base> make_collect_list_aggregation(
 
+/**
+ * @brief Factory to create a MERGE_LISTS aggregation.
+ *
+ * This aggregation is similar to `COLLECT_LIST` with the following differences:
+ *  - It requires the input values to be a non-nullable lists column, and
+ *  - The values (lists) corresponding to the same key will not result in a list of lists as output
+ *    from `COLLECT_LIST`. Instead, those lists will result in a list generated by merging them
+ *    together.
+ *
+ * In practice, this aggregation is used to merge the partial results of multiple (distributed)
+ * groupby `COLLECT_LIST` aggregations into a final `COLLECT_LIST` result. Those distributed
+ * aggregations were executed on different values columns partitioned from the original values
+ * column, then their results were (vertically) concatenated before given as the values column for
+ * this aggregation.
+ *
+ * Note that this aggregation does not accept any null handling parameter as it is designed for
+ * simply merging lists of the input lists column. Because the output from `COLLECT_LIST` are
+ * non-nullable, the input lists column to this aggregation is also required to be non-nullable (but
+ * its child column containing list entries is not subject to this requirement).
+ */
+template <typename Base = aggregation>
+std::unique_ptr<Base> make_merge_lists_aggregation();
+
+/**
+ * @brief Factory to create a MERGE_SETS aggregation.
+ *
+ * This aggregation is similar to `COLLECT_SET` with the following differences:
+ *  - It requires the input values to be a non-nullable lists column, and
+ *  - The values (lists) corresponding to the same key will result in a list generated by merging
+ *    them together then dropping duplicate entries.
+ *
+ * In practice, this aggregation is used to merge the partial results of multiple (distributed)
+ * groupby `COLLECT_LIST/COLLECT_SET` aggregations into a final `COLLECT_SET` result. Those
+ * distributed aggregations were executed on different values columns partitioned from the original
+ * values column, then their results were (vertically) concatenated before given as the values
+ * column for this aggregation.
+ *
+ * This aggregation firstly calls `MERGE_LISTS` to merge the input lists into intermediate lists,
+ * then it calls `lists::drop_list_duplicates` on them to remove duplicate list entries. As such,
+ * the input (partial results) to this aggregation should be generated by (distributed)
+ * `COLLECT_LIST` aggregations, not `COLLECT_SET`, to avoid unnecessarily removing duplicate entries
+ * for the partial results.
+ *
+ * Similar to `MERGE_LISTS`, this aggregation does not need the `null_policy` parameter and requires
+ * the input lists column to be non-nullable (its child column containing list entries is not
+ * subject to this requirement).
+ *
+ * Since duplicate list entries will be removed, the parameters `null_equality` and `nan_equality`
+ * are needed for calling to `lists::drop_list_duplicates`.
```

Comment:
These are all implementation details, and should not be in the public documentation. Instead, please clearly document the semantics and parameters visible to the caller.

The implementation details comments can be moved to the implementation, where they will be more helpful.

Reply:
Thanks. I have moved the detailed doxygen to detail implementation (https://github.com/rapidsai/cudf/pull/8436/files#diff-a943ede169c709882f6825c90823e342e974bbb95ef1a5ebe1bea7f083caf811R407).